pkg/controller/seed-controller-manager/kubernetes/resources.go

```diff
@@ -120,7 +104,9 @@ func (r *Reconciler) ensureResourcesAreDeployed(ctx context.Context, cluster *ku
 	}
 
 	// check that all StatefulSets are created
-	if err := r.ensureStatefulSets(ctx, cluster, data); err != nil {
+	if ok, err := r.statefulSetHealthCheck(ctx, cluster); !ok || err != nil {
+		r.log.Info("Skipping reconcile for StatefulSets, not healthy yet")
```

Comment:
Does this affect all StatefulSets? Even Prometheus?

Reply:
No, Prometheus is managed by a separate controller. The StatefulSets created by this `ensureStatefulSets` are defined here: https://github.com/kubermatic/kubermatic/blob/4c0ac9c42f60a54fbf27da8b03086abd4f10e154/pkg/controller/seed-controller-manager/kubernetes/resources.go#L535. It's only etcd and - optionally - rancher-server.